### PR TITLE
Improve uses of get, fix temperature bug

### PR DESCRIPTION
**Pull request by pamelafox**
This pull request primarily focuses on improving the handling of dictionary `get` method calls in the codebase. The changes involve replacing the `or` operator with the default parameter of the `get` method to provide cleaner and more Pythonic code. 

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[X] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Automated tests

### CODE REVIEW

File: app/backend/approaches/chatreadretrievereadvision.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ async def run_until_final_call(
         chat_completion: ChatCompletion = await self.openai_client.chat.completions.create(
             model=self.gpt4v_deployment if self.gpt4v_deployment else self.gpt4v_model,
             messages=messages,
-            temperature=overrides.get("temperature") or 0.0,
```
Comment (pamelafox) [inline app/backend/approaches/chatreadretrievereadvision.py:113]: This was a bug, temperature parameter shouldn't be used in the first call.

File: app/backend/approaches/retrievethenread.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ async def run(
                 # Azure Open AI takes the deployment name as the model name
                 model=self.chatgpt_deployment if self.chatgpt_deployment else self.chatgpt_model,
                 messages=message_builder.messages,
-                temperature=overrides.get("temperature") or 0.3,
```
Comment (pamelafox) [inline app/backend/approaches/retrievethenread.py:121]: Interesting that our default is 0.3 for RTR, didnt realize thst.

File: scripts/prepdocslib/filestrategy.py [hunk 3]
```diff
@@ -59,8 +59,8 @@ async def run(self, search_info: SearchInfo):
             async for file in files:
                 try:
                     key = file.file_extension()
-                    processor = self.file_processors[key]
```
Comment (pamelafox) [inline scripts/prepdocslib/filestrategy.py:62]: This was a bug, this needs to be a get.

File: app/backend/approaches/approach.py [hunk 4]
```diff
@@ -96,7 +96,7 @@ def __init__(
         self.openai_host = openai_host
 
     def build_filter(self, overrides: dict[str, Any], auth_claims: dict[str, Any]) -> Optional[str]:
-        exclude_category = overrides.get("exclude_category") or None
+        exclude_category = overrides.get("exclude_category")
```
Comment (mattgotteiner) [inline app/backend/approaches/approach.py:99]: right .get defaults to None as the returned value if it's not present. I think I helped copy paste this everywher sorry about that

File: app/backend/approaches/chatreadretrievereadvision.py (hunk 1 above)
Comment (mattgotteiner) [reply app/backend/approaches/chatreadretrievereadvision.py:113]: Is there a doc that indicates this? Would be helpful to add it in a comment for folks reading this

File: app/backend/approaches/retrievethenread.py (hunk 2 above)
Comment (mattgotteiner) [reply app/backend/approaches/retrievethenread.py:121]: I think these choices were made a long time ago and never revisited. Perhaps we can use the evaluation repo on the samples to figure out which one is the best :)

File: app/backend/approaches/retrievethenread.py [hunk 5]
```diff
@@ -118,7 +118,7 @@ async def run(
                 # Azure Open AI takes the deployment name as the model name
                 model=self.chatgpt_deployment if self.chatgpt_deployment else self.chatgpt_model,
                 messages=message_builder.messages,
```
Comment (mattgotteiner) [inline app/backend/approaches/retrievethenread.py:120]: I sometimes see issues related to "output isn't consistent". In a follow up PR would love to see us using the seed parameter and let customers put it in developer settings

https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/reproducible-output?tabs=pyton

File: scripts/prepdocslib/filestrategy.py (hunk 3 above)
Comment (mattgotteiner) [reply scripts/prepdocslib/filestrategy.py:62]: great catch!

**Review by mattgotteiner — APPROVED** [review]

File: app/backend/approaches/chatreadretrievereadvision.py (hunk 1 above)
Comment (pamelafox) [reply app/backend/approaches/chatreadretrievereadvision.py:113]: Sure, added comment to both chat files.

File: app/backend/approaches/retrievethenread.py (hunk 2 above)
Comment (pamelafox) [reply app/backend/approaches/retrievethenread.py:121]: Yep good idea! I've documented for now.

**Pull request merged**